manager/controllers/app/select_modules.go

```diff
@@ -338,18 +343,21 @@ func supportsSourceInterface(edge *Edge, source *Node) bool {
 		}
 		hasSources = true
 		// connection via Source
-		if match(inter.Source, source.Connection) {
+		if source != nil && match(inter.Source, source.Connection) {
```

Comment:
I'd consider to change the name, otherwise, it looks weird when you are comparing `Source` and `source.Connections` 
maybe instead of `source` use `sourceNode` ?

Reply:
Changed.